dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/RelationshipTypeControllerTest.java

```diff
@@ -0,0 +1,109 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller;
+
+import static org.hisp.dhis.web.WebClientUtils.assertStatus;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+
+import org.hisp.dhis.jsontree.JsonList;
+import org.hisp.dhis.jsontree.JsonString;
+import org.hisp.dhis.web.HttpStatus;
+import org.hisp.dhis.webapi.DhisControllerIntegrationTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Zubair Asghar
+ */
+
+class RelationshipTypeControllerTest extends DhisControllerIntegrationTest
+{
+
+    private String program;
+
+    private String attrA;
+
+    private String attrB;
+
+    private String attrC;
+
+    private String trackedEntityType;
+
+    @BeforeEach
+    void setUp()
+    {
+        trackedEntityType = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityTypes/",
+                "{'name':'person', 'shortName':'person', 'description':'person', 'allowAuditLog':false }" ) );
+
+        attrA = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityAttributes/",
+                "{'name':'attrA', 'shortName':'attrA', 'valueType':'TEXT', 'aggregationType':'NONE'}" ) );
+
+        attrB = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityAttributes/",
+                "{'name':'attrB', 'shortName':'attrB', 'valueType':'TEXT', 'aggregationType':'NONE'}" ) );
+
+        attrC = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityAttributes/",
+                "{'name':'attrC', 'shortName':'attrC', 'valueType':'TEXT', 'aggregationType':'NONE'}" ) );
+
+        program = assertStatus( HttpStatus.CREATED, POST( "/programs/",
+            "{'name':'test program', 'id':'VoZMWi7rBgj', 'shortName':'test program','programType':'WITH_REGISTRATION', 'trackedEntityType': {"
+                +
+                "'id': '" + trackedEntityType + "'}," +
+                " 'programTrackedEntityAttributes' :  [{ 'trackedEntityAttribute' :{'id': '" + attrA
+                + "' }}, { 'trackedEntityAttribute' :{'id': '" + attrB + "' }}, { 'trackedEntityAttribute' :{'id': '"
+                + attrC + "' }}] }" ) );
+
+    }
+
+    @Test
+    void testPostingRelationshipTypes()
+    {
+        String relationshipTypeId = assertStatus( HttpStatus.CREATED,
```

Comment:
We agreed on a test naming convention like shouldXXXXWhenXXXX

Reply:
resolved